src/android/ContactAccessorSdk5.java

```diff
@@ -307,24 +306,41 @@ public JSONArray search(JSONArray fields, JSONObject options) {
      * @return     a JSONObject representing the contact
      * @throws JSONException
      */
+    @Override
     public JSONObject getContactById(String id) throws JSONException {
-        // Call overloaded version with no desiredFields
-        return getContactById(id, null);
+        Cursor c = mApp.getActivity().getContentResolver().query(
+                ContactsContract.Data.CONTENT_URI,
+                null,
+                ContactsContract.Data.CONTACT_ID + " = ? ",
+                new String[] { id },
+                ContactsContract.Data.CONTACT_ID + " ASC"
+        );
+        return getContactById(c);
```

Comment:
This code is a duplicate of `getContactByRawId` except for the `ContactsContract.Data` value. Couldn't we reuse it?

Reply:
Yeah I suppose we could add the the row id instead as an argument to the common method and perform the query with it.

Reply:
Done in 08b24f687c231415dfe9ce74257d808a7dd0ffe7